zospy/zpcore.py

```diff
@@ -391,21 +410,31 @@ def connect_as_extension(self, instancenumber: int = 0) -> bool:
 
         self.Application = self.Connection.ConnectAsExtension(instancenumber)
 
-        if self.Application.IsValidLicenseForAPI:
-            return True
-        else:
+        if not self.Application.IsValidLicenseForAPI:
             logger.critical("OpticStudio Licence is not valid for API, connection not established")
+            raise ConnectionRefusedError("OpticStudio Licence is not valid for API, connection not established")
             return False
-
-    def create_new_application(self) -> bool:
+        else:
+            if not return_primary_system:
+                return True
+            else:
+                return self.get_primary_system()
+    
+    def create_new_application(self, return_primary_system: bool = False) -> bool | OpticStudioSystem:
```

Comment:
The suggestions for the docstring of `connect_as_extension` go for this one as well.